grl/environment/wrappers/gamma_terminal.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+from typing import Union, Tuple
+from grl.mdp import MDP
+import gymnasium as gym
+
+class GammaTerminalWrapper(gym.Wrapper):
+    """
+    A wrapper for a gymnasium.Env that makes it compatible with our step() API,
+    in particular the gamma_terminal feature. 
+    """
+    def __init__(self, env: Union[gym.Env, gym.Wrapper], gamma) -> None:
+        super().__init__(env)
+        assert not isinstance(self.env, MDP)
+        self.gamma = gamma
+
+    def step(self, action: int, gamma_terminal: bool, **kwargs) -> Tuple[np.ndarray, float, bool, bool, dict]:
+        observation, reward, terminal, truncated, info = self.env.step(action, **kwargs)
+        if gamma_terminal:
```

Comment:
working on the above comment, but this shouldn't be a bug (and is in fact how it was originally implemented in `MDP`) because it doesn't touch `terminal` in the case we don't gamma-terminate. In other words, if `step()` already returned `terminal=True`, then the rest of the function will have no effect.